### DISCUSSION TITLE
next/image is using inline styles, any way to disable it?

### DISCUSSION THREAD

**Discussion by PassTheMayo** [vercel/next.js#47966] [Help]
### Summary

I've noticed while using an SEO auditing tool that there appears to be inline styles on my page. This confused me at first because I use TailwindCSS for everything, but while looking through the static HTML sent by the server I noticed that there is a `<img style="color:transparent;">` tag, which in the source code is a `next/image` component. Is there any way that I can disable the inline style for this component since it does not affect anything?

### Additional information

```js
Operating System:
      Platform: darwin
      Arch: x64
      Version: Darwin Kernel Version 22.3.0: Mon Jan 30 20:42:11 PST 2023; root:xnu-8792.81.3~2/RELEASE_X86_64
    Binaries:
      Node: 16.19.0
      npm: 8.19.3
      Yarn: N/A
      pnpm: N/A
    Relevant packages:
      next: 13.2.5-canary.30
      eslint-config-next: N/A
      react: 18.2.0
      react-dom: 18.2.0
```

**Comment by frattaro** [comment]
yep, i get a console error from the CSP blocking it

**Comment by j-mendez** [comment]
@PassTheMayo a temp workaround, if you use `style={{ color: undefined }}` it will overwrite the style causing the removal of the inline styles.

**Comment by pheroMona13** [comment]
This component is currently unusable, in my opinion. I won't compromise my website's security (CSP header) just for an Image element to have a transparent style color!

**Reply by j-mendez** [reply]
@pheroMona13 if you use `style={{ color: undefined }}` it will remove it on the server and client allowing CSP to work. I think all of the inline styles should be opt in and allow a css mapping to custom classes as an alternative.

**Reply by pheroMona13** [reply]
@j-mendez This hack doesn't seem to be working on the next version 13.4.12.

**Reply by j-mendez** [reply]
I am using `13.4.12` and it seems to work for me. Attaching some screenshots to give me details. ![Screenshot of workaround to remove inline css style from next/image component](https://github.com/vercel/next.js/assets/8095978/b0f4e427-70a1-4598-b98f-3a735fdd34d9)

![security policy for https://a11ywatch.com/features](https://github.com/vercel/next.js/assets/8095978/4b670ba1-7672-4a29-8e37-a22644d1215a).

**Reply by j-mendez** [reply]
@pheroMona13 at https://a11ywatch.com/blog is another example of the removal using a jpg. Can confirm it works for gifs and jpgs, have not looked at the code for next/image on how it is handling the inline style or tested other images.  ![display of using `style={{ color: undefined }}` in order to remove inline styles for CSP  ](https://github.com/vercel/next.js/assets/8095978/6d234ea4-622f-4955-bb72-94126d8664e4)

**Reply by pheroMona13** [reply]
@j-mendez Thank you for taking the time to share your results and screenshots. However, I'm still unable to remove the inline style. I even created a new Next.js project on another machine and tested it with the default template, but I had no luck. It's possible that your solution depends on another configuration in your project.

**Reply by don-galak** [reply]
Have you tried with `style={{ color: null }}`? It seems to be working with `next: 14.1.3`.

**Comment by gberkouwer** [comment]
The original question is not fully answered, so here is a retry :-). is there a way to fully disable next/image using inline styles? So completely removing the 'style='?
